src/eom-transform.c

```diff
@@ -132,8 +132,8 @@ eom_transform_apply (EomTransform *trans, GdkPixbuf *pixbuf, EomJob *job)
 	}
 
 	/* create the resulting pixbuf */
-	dest_width = abs (dest_bottom_right.x - dest_top_left.x + 1);
-	dest_height = abs (dest_bottom_right.y - dest_top_left.y + 1);
+	dest_width = trunc (fabs (dest_bottom_right.x - dest_top_left.x + 1));
+	dest_height = trunc (fabs (dest_bottom_right.y - dest_top_left.y + 1));
```

Comment:
Why not simply this? Same result, and closer to what was actually happening before: we just acknowledge that the value is truncated.
```c
	dest_width = fabs ((int) (dest_bottom_right.x - dest_top_left.x + 1));
	dest_height = fabs ((int) (dest_bottom_right.y - dest_top_left.y + 1));
```

Reply:
Your proposal didn't remove the build warning -Wabsolute-value.
```bash
#!/bin/bash
CFLAGS="-g -O -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wendif-labels -Wformat=2 -Winit-self -Wswitch-enum -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-align -Wwrite-strings -Wjump-misses-init -Wlogical-op -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wnested-externs -Wundef -Wformat-security" ./autogen.sh --prefix=/usr
make clean &> /dev/null
make 2> report.txt
for warning in $(grep warning report.txt | grep -Po '\[\-W.*\]' | sed 's/[][]//g' | sort -u)
do
 type=${warning#"-W"}
 num=$(echo "grep $type report.txt | wc -l" | sh)
 echo "$warning $num"
done
```
```patch
diff --git a/src/eom-transform.c b/src/eom-transform.c
index 54fe440..010ca30 100644
--- a/src/eom-transform.c
+++ b/src/eom-transform.c
@@ -132,8 +132,8 @@ eom_transform_apply (EomTransform *trans, GdkPixbuf *pixbuf, EomJob *job)
        }
 
        /* create the resulting pixbuf */
-       dest_width = trunc (fabs (dest_bottom_right.x - dest_top_left.x + 1));
-       dest_height = trunc (fabs (dest_bottom_right.y - dest_top_left.y + 1));
+       dest_width = fabs ((int) (dest_bottom_right.x - dest_top_left.x + 1));
+       dest_height = fabs ((int) (dest_bottom_right.y - dest_top_left.y + 1));
 
        dest_pixbuf = gdk_pixbuf_new (GDK_COLORSPACE_RGB,
                               gdk_pixbuf_get_has_alpha (pixbuf),
```
```diff
1551a1552,1558
> eom-transform.c: In function ‘eom_transform_apply’:
> eom-transform.c:135:15: warning: using floating point absolute value function ‘fabs’ when argument is of integer type ‘int’ [-Wabsolute-value]
>   135 |  dest_width = fabs ((int) (dest_bottom_right.x - dest_top_left.x + 1));
>       |               ^~~~
> eom-transform.c:136:16: warning: using floating point absolute value function ‘fabs’ when argument is of integer type ‘int’ [-Wabsolute-value]
>   136 |  dest_height = fabs ((int) (dest_bottom_right.y - dest_top_left.y + 1));
>       |                ^~~~
```

Reply:
abs() instead of fabs()?
```patch
diff --git a/src/eom-transform.c b/src/eom-transform.c
index 54fe440..19a6c2d 100644
--- a/src/eom-transform.c
+++ b/src/eom-transform.c
@@ -132,8 +132,8 @@ eom_transform_apply (EomTransform *trans, GdkPixbuf *pixbuf, EomJob *job)
        }
 
        /* create the resulting pixbuf */
-       dest_width = trunc (fabs (dest_bottom_right.x - dest_top_left.x + 1));
-       dest_height = trunc (fabs (dest_bottom_right.y - dest_top_left.y + 1));
+       dest_width = abs ((int) (dest_bottom_right.x - dest_top_left.x + 1));
+       dest_height = abs ((int) (dest_bottom_right.y - dest_top_left.y + 1));
 
        dest_pixbuf = gdk_pixbuf_new (GDK_COLORSPACE_RGB,
                               gdk_pixbuf_get_has_alpha (pixbuf),
```

Reply:
> abs() instead of fabs()?

Oops yes that's what I meant to suggest

Reply:
Done